src/cider/nrepl/middleware/info.clj

```diff
@@ -280,22 +281,27 @@
                             (sort-by count))
     :else (:arglists info)))
 
-(defn format-eldoc [raw-eldoc]
-  (map #(mapv str %) raw-eldoc))
+(defn format-arglists [raw-arglists]
+  (map #(mapv str %) raw-arglists))
 
-(defn eldoc
-  [info]
-  (if-let [raw-eldoc (extract-eldoc info)]
-    (format-eldoc raw-eldoc)))
+(defn format-class
+  [raw-class]
+  (clojure.string/join ", " (map str raw-class)))
+
+(defn extract-class
+  [{:keys [class candidates] :as info}]
+  (cond
+    class (str class)
+    candidates (format-class (map key candidates))))
 
 (defn eldoc-reply
   [msg]
-  (let [info (info msg)
-        var-eldoc (eldoc info)]
-    (if var-eldoc
-      {:eldoc var-eldoc
+  (let [info (info msg)]
+    (if-let [arglists (extract-arglists info)]
+      {:eldoc (format-arglists arglists)
        :name (:name info)
-       :ns (str (:ns info))}
+       :ns (str (:ns info))
+       :class (extract-class info)}
```

Comment:
Guess it'd be best if clojure symbols didn't have the `class` key and java stuff didn't have `ns`. Btw, `class` can also be an interface, but I guess this is not very important.